cardano-crypto-tests/src/Test/Crypto/DSIGN.hs

```diff
@@ -369,3 +677,140 @@ testEcdsaWithHashAlgorithm _ name = adjustOption defaultTestEnough . testGroup n
   where
     genMsg :: Gen MessageHash
     genMsg = hashAndPack (Proxy @h) . messageBytes <$> arbitrary
+#endif
+
+prop_dsignm_verify_neg_msg
+  :: forall v. (DSIGNMAlgorithm IO v, ContextDSIGNM v ~ (), SignableM v Message)
+  => Lock
+  -> Proxy v
+  -> Message
+  -> Message
+  -> PinnedSizedBytes (SeedSizeDSIGNM v)
+  -> Property
+prop_dsignm_verify_neg_msg lock _ a a' =
+  ioPropertyWithSK @v lock $ \sk -> do
+    sig <- signDSIGNM () a sk
+    vk <- deriveVerKeyDSIGNM sk
+    return $
+      a /= a' ==> verifyDSIGNM () vk a' sig =/= Right ()
+--
+-- Libsodium
+--
+
+-- TODO: use these
```

Comment:
Are those functions below gonna be finished up in a subsequent PR?

Reply:
Possibly, but not in this exact form. I'll remove, and wrote issue #363 so we don't forget.

Reply:
Perfect